lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,894 @@
+        self.ice_cube.coord('threshold').points = [0.5, 1., 4.]
+        msg = (r"No matching prob\(Ice\) cube for threshold 2.")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            self.plugin.apply_ice(self.fg_cube, self.ice_cube)
+
+    def test_ice_null(self):
+        """Test that small VII probs do not increase lightning risk"""
+        self.ice_cube.data[:, 7, 7] = 0.
+        self.ice_cube.data[0, 7, 7:9] = 0.5
+        self.fg_cube.data[0, 7, 7] = 0.25
+        expected = self.fg_cube.copy()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.25
+        result = self.plugin.apply_ice(self.fg_cube,
+                                       self.ice_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_ice_zero(self):
+        """Test that zero VII probs do not increase lightning risk"""
+        self.ice_cube.data[:, 7, 7] = 0.
+        self.fg_cube.data[0, 7, 7] = 0.
+        expected = self.fg_cube.copy()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.
+        result = self.plugin.apply_ice(self.fg_cube,
+                                       self.ice_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_ice_small(self):
+        """Test that small VII probs do increase lightning risk"""
+        self.ice_cube.data[:, 7, 7] = 0.
+        self.ice_cube.data[0, 7, 7] = 0.5
+        self.fg_cube.data[0, 7, 7] = 0.
+        expected = self.fg_cube.copy()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.05
+        result = self.plugin.apply_ice(self.fg_cube,
+                                       self.ice_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_ice_large(self):
+        """Test that large VII probs do increase lightning risk"""
+        self.ice_cube.data[:, 7, 7] = 1.
+        self.fg_cube.data[0, 7, 7] = 0.
+        expected = self.fg_cube.copy()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.9
+        result = self.plugin.apply_ice(self.fg_cube,
+                                       self.ice_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_ice_large_long_fc(self):
+        """Test that large VII probs do not increase lightning risk when
+        forecast lead time is large"""
+        self.ice_cube.data[:, 7, 7] = 1.
+        self.fg_cube.data[0, 7, 7] = 0.
+        self.fg_cube.coord('forecast_period').points = [3.]  # hours
+        expected = self.fg_cube.copy()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.0
+        result = self.plugin.apply_ice(self.fg_cube,
+                                       self.ice_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+
+class Test_process(IrisTest):
```

Comment:
I'm not sure you have tests that are testing the cube has the right dimension coordinates, the right metadata etc.

Reply:
I have added a couple of checks to the test_basic tests.